### ISSUE TITLE
dislocker in fstab with password

### ISSUE THREAD

**Issue by junadlao** [Aorimn/dislocker#196]
Can you please add the capability of being able to put the password in a separate (root-only accessible) file?... Instead of only having the option to put in in fstab for everybody to read. Thanks!

**Comment by thosoo** [comment]
This could be solved by a similar approach like mounting a curlftpfs in fstab by having a /root/.dislocker file which only root can read.

**Comment by junadlao** [comment]
Yes. Reading a /root/.dislocker file would be a great solution. As of now, we can't really chmod 600 the fstab file and has to leave it universally accessible and thus present our password to everybody.

**Comment by hk3wx** [comment]
I'm looking forward to use a JSON file with the password ;D

**Comment by szaimen** [comment]
Would love to see this as well

**Comment by Aorimn** [comment]
This can already be done by creating a BEK file protector on Windows.

First use manage-bde to create this protector, e.g. something like `manage-bde -protectors -add c: -sk PATH_WHERE_TO_WRITE_THE_BEK_FILE`.
Get this file which extension is `.BEK` and use it with dislocker's option `-f` (or use `bekfile=/path/to/bekfile` in the fstab).
Finally, make sure the permissions on the BEK file are secured.

As the use case seems to be covered, I'm going to close this. Please tell me if the solution doesn't suit your needs by describing your constraints on a follow-up comment and reopening this issue.

**Comment by junadlao** [comment]
Sorry to reply so late... but as my last post on this topic, let me just share the solution that I happen to stumble upon.
As I recently upgraded an old system to Fedora 33, I was surprised when I attached a bitlocker-protected external drive because the freshly-built system suddenly recognized it and asked for the bitlocker passphrase. I was actually expecting it to not do anything as by then I was just about preparing to install this dislocker project. So -- to my surprise, bitlocker-unlocking is now a default feature of Fedora 33 and it is done thru the default linux encryption system of LUKS, dm-crypt and cryptsetup which was all installed as I always encrypt my whole system anyway. All I needed to do to add automatic bitlocker-decryption is to add an entry to the /etc/crypttab and I can specify an external file there that would contain my passphrase with the usual keyfile="/path/to/file" entry. What an elegant solution it turned out to be!
